framework/DataObjects/DataSet.py

```diff
@@ -0,0 +1,1853 @@
+        return False
+      # check if index-dependent variables have matching shapes
+      # FIXME: this check will not work in case of variables depending on multiple indexes. When this need comes, we will change this check(alfoa)
+      if self.indexes:
+        if key in self._fromVarToIndex and rlz[self._fromVarToIndex[key]].shape != rlz[key].shape:
+          self.raiseAWarning('Variable "{}" has not a consistent shape with respect its index "{}": shape({}) /= shape({})!'.format(key,self._fromVarToIndex[key],rlz[key].shape,rlz[self._fromVarToIndex[key]].shape))
+          return False
+    return okay
+
+  def _clearAlignment(self):
+    """
+      Clears the alignment tracking for the collector, and removes columns from it if necessary
+      @ In, None
+      @ Out, None
+    """
+    # get list of indexes that need to be removed since we're starting over with alignment
+    toRemove = list(self._orderedVars.index(var) for var in self.indexes if (var not in self._alignedIndexes.keys() and var in self._orderedVars))
+    # sort them in reverse order so we don't screw up indexing while removing
+    toRemove.sort(reverse=True)
+    for index in toRemove:
+      self._orderedVars.pop(index)
+      self._collector.removeEntity(index)
+    self._alignedIndexes = {}
+
+  def _clearParentEndingStatus(self,rlz):
+    """
+      If self is hierarchical, then set the parent of the given realization "rlz" to False.
+      @ In, rlz, dict, realization (from addRealization, already formatted)
+      @ Out, None
+    """
+    # TODO set global status of 'parentID' instead of check every time
+    idVar = 'RAVEN_parentID'
+    endVar = 'RAVEN_isEnding'
+    if idVar in self.getVars():
+      # get the parent ID
+      parentID = rlz[idVar]
+      # if root or parentless, nothing to do
+      if parentID == "None":
+        return
+      # otherwise, find the index of the match
+      idx,match = self.realization(matchDict={'prefix':parentID})
+      self._changeVariableValue(idx,endVar,False)
+
+  def _collapseNDtoDataArray(self,data,var,labels=None,dtype=None):
+    """
+      Converts a row of numpy samples (float or xr.DataArray) into a single DataArray suitable for a xr.Dataset.
+      @ In, data, np.ndarray, array of either float or xr.DataArray; array must be single-dimension
+      @ In, var, str, name of the variable being acted on
+      @ In, labels, list, optional, list of labels to use for collapsed array under self.sampleTag title
+      @ In, dtype, type, optional, type from _getCompatibleType to cast data as
+      @ Out, DataArray, xr.DataArray, single dataarray object
+    """
+    assert(isinstance(data,np.ndarray))
+    assert(len(data.shape) == 1)
+    # set up sampleTag values
+    if labels is None:
+      labels = range(len(data))
+    else:
+      assert(len(labels) == len(data))
+    # find first non-None entry, and get its type if needed
+    dataType = type(None)
+    i = -1
```

Comment:
negative numbers get such a bad rap.  fixed.

Reply:
On second read, this would take an extra line of code, since "i" would be `++` one too many times, and I'd have to subtract it back off, or some similarly excessive change.  Reverting.